packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -9137,17 +9205,55 @@ describe('SnapController', () => {
       );
 
       const state = { foo: 'bar' };
+
+      const promise = waitForStateChange(messenger);
       await messenger.call(
         'SnapController:updateSnapState',
         MOCK_SNAP_ID,
         state,
         true,
       );
 
+      await promise;
+
       expect(pbkdf2Sha512).toHaveBeenCalledTimes(1);
 
       snapController.destroy();
     });
+
+    it('logs an error message if the state fails to persist', async () => {
```

Comment:
Can we add a test that the persistence is properly mutex and therefore queued running two parallel update calls or something